src/main/java/spoon/SpoonModelBuilder.java

```diff
@@ -76,9 +79,15 @@
 	 */
 	boolean build(JDTBuilder builder);
 
-	/** The types of compilable elements */
-	enum InputType {
-		FILES, CTTYPES
+	/** The types of compilable elements
+	 * FILES - compiles the java files from the file system, which were registered by {@see SpoonModelBuilder#addInputSource()} and {@see SpoonModelBuilder#addTemplateSource(File)}
+	 * CTTYPES - compiles virtual java files, which are dynamically generated from the all top level classes of the CtModel by {@see DefaultJavaPrettyPrinter}
+	 */
+	interface InputType {
+		InputType FILES = FileCompilerConfig.INSTANCE;
+		InputType CTTYPES = FactoryCompilerConfig.INSTANCE;
+
+		void initializeCompiler(JDTBatchCompiler compiler);
```

Comment:
add javadoc:
/** responsible for setting the parameters of JDTBatchCompiler, must call setCompilationUnits() */